sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/ExtractBenchmark.scala

```diff
@@ -84,24 +86,36 @@ object ExtractBenchmark extends SqlBasedBenchmark {
     }
   }
 
+  private case class Settings(fields: Seq[String], func: Seq[String], iterNum: Long)
+
   override def runBenchmarkSuite(mainArgs: Array[String]): Unit = {
     val N = 10000000L
-    val fields = Seq(
+    val datetimeFields = Seq(
       "MILLENNIUM", "CENTURY", "DECADE", "YEAR",
       "ISOYEAR", "QUARTER", "MONTH", "WEEK",
       "DAY", "DAYOFWEEK", "DOW", "ISODOW",
       "DOY", "HOUR", "MINUTE", "SECOND",
       "MILLISECONDS", "MICROSECONDS", "EPOCH")
+    val intervalFields = Seq(
+      "MILLENNIUM", "CENTURY", "DECADE", "YEAR",
+      "QUARTER", "MONTH", "DAY",
+      "HOUR", "MINUTE", "SECOND",
+      "MILLISECONDS", "MICROSECONDS", "EPOCH")
+    val settings = Map(
```

Comment:
not a big deal but `Settings` seems only used within `runBenchmarkSuite`. I think it's fine to just make this pretty with indentation.

Reply:
? Could you clarify, please. Do you want to replace `Settings` by let's say tuples?

Reply:
This one?
```scala
    for {
      (dataType, (fields, funcs, iterNum)) <- Map(
        "timestamp" -> (datetimeFields, Seq("extract", "date_part"), N),
        "date" -> (datetimeFields, Seq("extract", "date_part"), N),
        "interval" -> (intervalFields, Seq("date_part"), N))
      func <- funcs} {
```

Reply:
yup, I was thinking like that. I don't mind if you prefer the current way. no biggie.